shared/string.js

```diff
@@ -0,0 +1,27 @@
+String.prototype.isEmpty = function() {
+  return this.trim() === '';
+};
+
+String.prototype.isBlankOrGitOrBit = function() {
+  return /^$|github.com|bitbucket.org/g.test(this);
+};
+
+String.prototype.toLowerHyphenated = function() {
```

Comment:
That's not very DRY, because it would mean repeating the lowercase and trim logic. This function provides the exact sequence we will want to reuse in various places.